eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -324,25 +322,112 @@ function Invoke-Pack() {
+    }
+    
+    $packagesToGen = @()
+    $newpackageoutput = "newPackageOutput.json"
+    if ( $serviceType -eq "resource-manager" ) {
+        Write-Host "Generate resource-manager SDK client library."
+        $package = $service
+        if ($packageNameHash[$service] -ne "") {
+            $package = $packageNameHash[$service]
         }
-        if ($readmeFile -match $readmeFileRegex)
-        {
-            return $matches["specName"], $matches["serviceType"]
+        New-MgmtPackageFolder -service $service -packageName $package -sdkPath $sdkRootPath -commitid $commitid -readme $readmeFile -outputJsonFile $newpackageoutput
+        if ( !$?) {
+            Write-Error "Failed to create sdk project folder. exit code: $?"
+            exit 1
+        }
+        $newpackageoutputJson = Get-Content $newpackageoutput | Out-String | ConvertFrom-Json
+        $packagesToGen = $packagesToGen + @($newpackageoutputJson)
+        Remove-Item $newpackageoutput
+    } else {
+        Write-Host "Generate data-plane SDK client library."
+        npx autorest --version=3.7.3 --csharp $readmeFile --csharp-sdks-folder=$sdkRootPath --skip-csproj --clear-output-folder=true
+        $serviceSDKDirectory = (Join-Path $sdkPath sdk $service)
+        $folders = Get-ChildItem $serviceSDKDirectory -Directory -exclude *.*Management*,Azure.ResourceManager*
+        $folders |ForEach-Object {
+            $folder=$_.Name
+            New-DataPlanePackageFolder -service $service -namespace $folder -sdkPath $sdkRootPath -readme $readmeFile -outputJsonFile $newpackageoutput
+            $newpackageoutputJson = Get-Content $newpackageoutput | Out-String | ConvertFrom-Json
+            $packagesToGen = $packagesToGen + @($newpackageoutputJson)
+            if ( !$? ) {
+                Write-Error "Failed to create sdk project folder. exit code: $?"
+                exit 1
+            }
+            Remove-Item $newpackageoutput
         }
-        
     }
-    catch
+
+    foreach ( $package in $packagesToGen )
     {
-        Write-Error "Error parsing readme info"
-        Write-Error $_
+        $projectFolder = $newpackageoutputJson.projectFolder
+        $path = $newpackageoutputJson.path
+        $service = $newpackageoutputJson.service
+        $packageName = $newpackageoutputJson.packageName
+        Write-Host "projectFolder:$projectFolder"
+
+        # Generate Code
+        Invoke-Generate -sdkfolder $projectFolder
+        if ( !$?) {
+            Write-Error "Failed to generate sdk. exit code: $?"
+            exit 1
+        }
+
+        # Build
+        Invoke-Build -sdkfolder $projectFolder
+        if ( !$? ) {
+            Write-Error "Failed to build sdk. exit code: $?"
+            exit 1
+        }
+
+        # pack
+        Invoke-Pack -sdkfolder $projectFolder
+        if ( !$? ) {
+            Write-Error "Failed to packe sdk. exit code: $?"
+            exit 1
+        }
+        # Generate APIs
+        Push-Location $sdkRootPath
```

Comment:
You don't need to do this here as well. Just call `"&$sdkRootPath/eng/scripts/Export-API.ps1" $service` below. Even if  you did have to set the PowerShell-based current working directory (note: this doesn't impact Windows' CWD, which is what any other process you start will use), your `Write-Error` below - if someone has their `$ErrorActionPreference = 'Stop'` will be terminal and not restore the previous location. You could immediately after pushing type `trap { pop-location }` but just avoid this pattern. It's fragile at best.